lib/onelogin/ruby-saml/response.rb

```diff
@@ -112,78 +151,327 @@ def status_message
         end
       end
 
-      # Conditions (if any) for the assertion to run
+      # Gets the Condition Element of the SAML Response if exists.
+      # (returns the first node that matches the supplied xpath)
+      # @return [REXML::Element] Conditions Element if exists
+      #
       def conditions
         @conditions ||= xpath_first_from_signed_assertion('/a:Conditions')
       end
 
+      # Gets the NotBefore Condition Element value.
+      # @return [Time] The NotBefore value in Time format
+      #
       def not_before
         @not_before ||= parse_time(conditions, "NotBefore")
       end
 
+      # Gets the NotOnOrAfter Condition Element value.
+      # @return [Time] The NotOnOrAfter value in Time format
+      #
       def not_on_or_after
         @not_on_or_after ||= parse_time(conditions, "NotOnOrAfter")
       end
 
-      def issuer
-        @issuer ||= begin
+      # Gets the Issuers (from Response and Assertion).
+      # (returns the first node that matches the supplied xpath from the Response and from the Assertion)
+      # @return [Array] Array with the Issuers (REXML::Element)
+      #
+      def issuers
+        @issuers ||= begin
+          issuers = []
           node = REXML::XPath.first(document, "/p:Response/a:Issuer", { "p" => PROTOCOL, "a" => ASSERTION })
```

Comment:
try this with the XPath `/p:Response/a:Issuer | /p:Response/a:Assertion/a:Issuer` so you can get rid of line 184-191 and implement:
```ruby
nodes.each do |node|
  issuers << node.text
end
```